src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -42,471 +44,524 @@ public static class SystemNamespaceType
                 .WithRequiredParameter("filePath", LanguageConstants.String, "The path to the file that will be loaded")
                 .WithReturnResultBuilder(LoadContentAsBase64ResultBuilder, LanguageConstants.String)
                 .WithEvaluator(StringLiteralFunctionReturnTypeEvaluator)
                 .WithVariableGenerator(StringLiteralFunctionVariableGenerator)
-                .Build(),
-            new FunctionOverloadBuilder("loadJsonContent")
+                .Build();
+            yield return new FunctionOverloadBuilder("loadJsonContent")
                 .WithGenericDescription($"Loads the specified JSON file as bicep object. File loading occurs during compilation, not at runtime.")
                 .WithRequiredParameter("filePath", LanguageConstants.String, "The path to the file that will be loaded")
                 .WithOptionalParameter("jsonPath", LanguageConstants.String, "JSONPath expression to narrow down the loaded file. If not provided, a root element indicator '$' is used")
                 .WithOptionalParameter("encoding", LanguageConstants.LoadTextContentEncodings, "File encoding. If not provided, UTF-8 will be used.")
                 .WithReturnResultBuilder(LoadJsonContentResultBuilder, LanguageConstants.Any)
                 .WithEvaluator(JsonContentFunctionReturnTypeEvaluator)
                 .WithVariableGenerator(JsonContentFunctionVariableGenerator)
-                .Build(),
+                .Build();
 
-            new FunctionOverloadBuilder("items")
+            yield return new FunctionOverloadBuilder("items")
                 .WithGenericDescription("Returns an array of keys and values for an object. Elements are consistently ordered alphabetically by key.")
                 .WithRequiredParameter("object", LanguageConstants.Object, "The object to return keys and values for")
                 .WithReturnResultBuilder(ItemsResultBuilder, GetItemsReturnType(LanguageConstants.String, LanguageConstants.Any))
-                .Build(),
-        }.ToImmutableArray();
+                .Build();
+
+            if (featureProvider.LambdasEnabled)
+            {
+                yield return new FunctionOverloadBuilder("flatten")
+                    .WithGenericDescription("Flattens an array of arrays into an array containing all of the individual array elements in the original order.")
+                    .WithVariableParameter("array", LanguageConstants.Array, 0, "TBD")
+                    .WithReturnType(LanguageConstants.Array)
+                    .Build();
+
+                yield return new FunctionOverloadBuilder("filter")
+                    .WithGenericDescription("Filters an array with a custom filtering function.")
+                    .WithRequiredParameter("array", LanguageConstants.Array, "The array to filter")
+                    .WithRequiredParameter("lambda", OneParamLambda(LanguageConstants.Any, LanguageConstants.Bool), "The filtering function",
```

Comment:
>lambda [](http://example.com/codeflow?start=44&length=6)

These arguments shouldn't be called `lambda`. The user can tell from the type that it's a lambda. I think we should make them reflect the purpose of the function. I think `filter` or `predicate` would be better.

Reply:
Good point!